docs/DeveloperGuide.md

```diff
@@ -1,38 +1,462 @@
+
+
+- **Step 4**: Next, the `Command` object will then call method `CommandAddClient#execute(...)` or `CommandAddProperty#execute(...)` which will add a new `Client` or `Property` type object created into their respective `clientList`/`propertyList`.
+
+
+- **Step 5**: Lastly, method `Ui#showClientAddedConfirmationMessage()` or `Ui#showPropertyAddedConfirmationMessage()` is called to notify user about the successful addition of new client or property. Also, method `Storage#addToClientFile` or `Storage#addToPropertyFile` is called to update their respective storage files.
+
+The following are simplified sequence diagrams of add feature for client and property:
+![Add Client Sequence Diagram](diagrams/AddClientSequenceDiagram.JPG)
+<p align="center">
+Add Client Sequence Diagram
+</p>
+
+![Add Property Sequence Diagram](diagrams/AddPropertySequenceDiagram.JPG)
+<p align="center">
+Add Property Sequence Diagram
+</p>
+
+---
+
+### Delete Client/Property feature
+The **delete client/property** mechanism involves the following classes: ```ParseDeleteClient```,
+```ParseDeleteProperty```, ```CommandDeleteClient```, ```CommandDeleteProperty```,
+```ClientList```, ```PropertyList``` and ```PairingList```.
+
+Given below is an example usage scenario and how the delete client/property behaves at each step.
+
+**Step 1:** The user executes ```delete -client ic/INDEX``` or ```delete -property ip/INDEX```.
+The ```ParseDeleteClient``` or ```ParseDeleteProperty``` class is called respectively and the format of the user
+input is checked for any incorrect formatting.
+
+**Step 2:** If there are no errors, ```CommandDeleteClient``` or ```CommandDeleteProperty``` is called respectively.
+The ```CommandDeleteClient#execute()``` or ```CommandDeleteProperty#execute()``` method is then called.
+
+**Step 3:** The ```ClientList#deleteClient()``` or ```PropertyList#deleteProperty()``` method is called which 
+removes the Client or Property with that specific index from their respective ArrayList.
+
+**Step 4:** Any pairings involving that specific Client or Property is also deleted using the
+```pairingList#deletePairing()``` method. A message showing all the deleted pairs is shown to the user.
+
+**Step 5:** The corresponding line(s) in the respective files are deleted. The method is shown in the Storage
+Implementation section.
+
+The following *class diagram* shows all the classes involved in the **delete client/property** operation
+and their relationships.
+
+![Delete Client/Property Class Diagram](diagrams/DeleteClientPropertyCD.png)
+
+The following *sequence diagram* shows how the **delete client** operation works, showcasing the
+```ClientList#deleteClient()``` method.
+
+![Delete Client Sequence Diagram](diagrams/DeleteClientSD.png)
+
+The following *sequence diagram* shows how the **delete property** operation works, showcasing the
+```PropertyList#deleteProperty()``` method.
+
+![Delete Property Sequence Diagram](diagrams/DeletePropertySD.png)
```

Comment:
Lifeline is not drawn fully, not sure if currentListSize-- should be included...